osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/main/resources/db/migration/V20200408162500000__Additional_device_fields.sql

```diff
@@ -9,6 +9,15 @@ BEGIN
 
   	ALTER TABLE iec60870_device ALTER COLUMN common_address SET DEFAULT 0;
 
+    ALTER TABLE iec60870_device
+      ADD CONSTRAINT iec60870_device_gateway_key
+        UNIQUE (gateway_device_identification, information_object_address);
```

Comment:
Doesn't this give issues when you have more than 1 gateway device (`both gateway_device_identification` and `information_object_address` being `null`). This was why I wondered if the constraint could be here or would have to be enforced with repository code.